src/common/snippets/include/snippets/lowered/linear_ir.hpp

```diff
@@ -49,7 +49,10 @@ class LinearIR {
 
     ExpressionPtr create_expression(const std::shared_ptr<Node>& n, const std::vector<PortConnectorPtr>& inputs);
 
-    static LinearIR::container deep_copy_range(LinearIR::container::const_iterator begin, LinearIR::container::const_iterator end);
+    std::shared_ptr<LinearIR> clone() const;
```

Comment:
And for other classes. Because sometimes we can work with only instances. I'd like to avoid situations aka `auto new_smth = *{smth->clone()}` like [there](https://github.com/openvinotoolkit/openvino/blob/f62fd778a5c51ef426138e19f70aa59f1800709d/src/common/snippets/src/op/subgraph.cpp#L494).

Reply:
Yes, I see your point, but LIR is a different story. The thing with LIR is that it's handled by value and passed to transformations by reference almost everywhere in the pipeline. So when we write `auto a = m_linear_ir`, we actually expect a shallow copy, not a deep one. For example here, in [op::Kernel](https://github.com/openvinotoolkit/openvino/blob/9fafcabb7c7b221821b0b77616d9967c8d13223f/src/common/snippets/include/snippets/op/kernel.hpp#L23-L26), we don't need to deep copy of the whole IR (which is really expensive). 
In fact, we need to deep copy LIR only once, and that's the place you linked. And it looks a bit unnatural just because `LinearIR::clone()` is aligned with the rest of the `clone` methods and returns `shared_ptr` of the object, while LIR (despite the rest of the classes) is managed by value, not by pointer.

Reply:
I got it, good explained! Thanks a lot!